tests/execution_engine/test_sqlalchemy_batch_data.py

```diff
@@ -155,26 +157,35 @@ def test_instantiation_with_unknown_dialect(sqlite_view_engine):
 
 @pytest.mark.unit
 def test_instantiation_with_temp_table_schema():
-    engine = Mock(spec=["dialect", "execute"])
-    execution_engine = Mock(spec=SqlAlchemyExecutionEngine, engine=engine)
 
     # not supported
-    engine.dialect.name = "sqlite"
-    SqlAlchemyBatchData(
+    engine = MockSaEngine(dialect=Dialect(dialect="sqlite"))
+    execution_engine = Mock(spec=SqlAlchemyExecutionEngine, engine=engine)
+    batch_data = SqlAlchemyBatchData(
         execution_engine=execution_engine,
         query="test_query",
         create_temp_table=True,
         temp_table_schema_name="test_schema",
     )
-    assert "test_schema" not in str(engine.execute.call_args[0][0])
+    created_table_name: str = batch_data._create_temporary_table(
+        temp_table_name="hello",
+        query="test_query",
+        temp_table_schema_name="test_schema",
+    )
+    assert "test_schema" not in created_table_name
 
     # supported
     for dialect in ["bigquery", "snowflake", "vertica"]:
         engine.dialect.name = dialect
-        SqlAlchemyBatchData(
+        batch_data = SqlAlchemyBatchData(
             execution_engine=execution_engine,
             query="test_query",
             create_temp_table=True,
             temp_table_schema_name="test_schema",
         )
-        assert "test_schema" in str(engine.execute.call_args[0][0])
+        created_table_name: str = batch_data._create_temporary_table(
```

Comment:
Is this created_table_name or the query used to create the table? If the query, I would suggest renaming this and the other instances to `query_to_create_temp_table` or similar.

Reply:
great catch. Fixed now.